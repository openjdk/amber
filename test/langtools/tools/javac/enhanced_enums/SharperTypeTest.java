/*
 * @test /nodynamiccopyright/
 * @bug 8177505
 * @summary testing the sharper type of enhanced enum constants
 * @compile/fail/ref=SharperTypeTest.out -XDrawDiagnostics SharperTypeTest.java
 */

public class SharperTypeTest {
    enum Primitive<X> {
        INT<Integer>(Integer.class, 0) {
            int mod(int x, int y) { return x % y; }
            int add(int x, int y) { return x + y; }
        },
        FLOAT<Float>(Float.class, 0f)  {
            long add(long x, long y) { return x + y; }
        };

        final Class<X> boxClass;
        final X defaultValue;

        Primitive(Class<X> boxClass, X defaultValue) {
           this.boxClass = boxClass;
           this.defaultValue = defaultValue;
        }
    }

    void m() {
        int zero_int = Primitive.INT.mod(50, 2);           // ok
        int zero_float = Primitive.FLOAT.mod(50, 2);       // error
    }
}
