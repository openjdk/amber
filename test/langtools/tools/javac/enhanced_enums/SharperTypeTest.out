SharperTypeTest.java:29:41: compiler.err.cant.resolve.location.args: kindname.method, mod, , int,int, (compiler.misc.location.1: kindname.variable, FLOAT, SharperTypeTest.Primitive<java.lang.Float>)
1 error
